crates/spin-js-engine/src/lib.rs

```diff
@@ -447,20 +457,18 @@ fn handle(request: Request) -> Result<Response> {
     process.set_property("env", env)?;
 
     global.set_property("process", process)?;
+    
+    let headers = context.object_value()?;
+    for (key, value) in request.headers().iter() {
+        let header_value =  context.value_from_str(&str::from_utf8(value.as_bytes()).unwrap().to_owned());
```

Comment:
I think we should probably just let the error bubble up here (which will avoid the need for the `if let` below).
```suggestion
        let header_value =  context.value_from_str(&str::from_utf8(value.as_bytes()).unwrap().to_owned())?;
```